carbonmark/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,149 @@
+import { cx } from "@emotion/css";
+import InfoOutlinedIcon from "@mui/icons-material/InfoOutlined";
+import KeyboardArrowLeftOutlinedIcon from "@mui/icons-material/KeyboardArrowLeftOutlined";
+import KeyboardArrowRightOutlinedIcon from "@mui/icons-material/KeyboardArrowRightOutlined";
+import { ProjectMap } from "components/pages/Project/ProjectMap";
+import { TextInfoTooltip } from "components/TextInfoTooltip";
+import useEmblaCarousel, { EmblaOptionsType } from "embla-carousel-react";
+import { useResponsive } from "hooks/useResponsive";
+import { CarouselImage, Project } from "lib/types/carbonmark";
+import { useCallback, useEffect, useState } from "react";
+import * as styles from "./styles";
+
+type CarouselProps = {
+  images: Array<CarouselImage>;
+  options?: EmblaOptionsType;
+  location?: Project["location"];
+};
+
+const Carousel: React.FC<CarouselProps> = (props) => {
+  const { isMobile } = useResponsive();
+  const [maxSlidesToShow] = useState(5);
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [images, setImages] = useState<Array<CarouselImage>>(props.images);
+  const [mainRef, mainApi] = useEmblaCarousel(props.options);
+  const [thumbsRef, thumbsApi] = useEmblaCarousel({
+    containScroll: "keepSnaps",
+    dragFree: true,
+  });
+
+  const onThumbClick = useCallback(
+    (index: number) => {
+      if (!mainApi || !thumbsApi) return;
+      mainApi.scrollTo(index);
+    },
+    [mainApi, thumbsApi]
+  );
+
+  const onPreviousClick = useCallback(() => {
+    if (!mainApi || !thumbsApi) return;
+    mainApi.scrollPrev();
+    thumbsApi.scrollPrev();
+  }, [mainApi, thumbsApi]);
+
+  const onNextClick = useCallback(() => {
+    if (!mainApi || !thumbsApi) return;
+    mainApi.scrollNext();
+    thumbsApi.scrollNext();
+  }, [mainApi, thumbsApi]);
+
+  const onSelect = useCallback(() => {
+    if (!mainApi || !thumbsApi) return;
+    setSelectedIndex(mainApi.selectedScrollSnap());
+    thumbsApi.scrollTo(mainApi.selectedScrollSnap());
+  }, [mainApi, thumbsApi, setSelectedIndex]);
+
+  useEffect(() => {
+    if (!mainApi) return;
+    onSelect();
+    mainApi.on("select", onSelect);
+    mainApi.on("reInit", onSelect);
+  }, [mainApi, onSelect]);
+
+  useEffect(() => {
+    if (props.location) {
+      const images = [...props.images];
+      images.unshift({
+        url: "/map-thumbnail.png",
+        caption: images?.[1]?.caption,
+      });
+      setImages(images);
```

Comment:
This does not need to happen in an effect, nor does it need state, just use an inline variable

```ts
const imagesWithThumb = [...props.images];
if (props.location) imagesWithThumb.unshift(mapThumbnail);
```

Reply:
Yeah agreed here - I originally had two separate pieces of data for thumbnails and the main slider as I was trying to only prepend the image for the thumbnails and was keeping the thumbnail data in state, but ended up using the same data for both and forgot to remove the effect and state variables.